source/agent/quic/e2e-test/patches/samplertcservice.js

```diff
@@ -0,0 +1,569 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Copyright (C) <2018> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is borrowed from lynckia/licode with some modifications.
+
+/*global require, __dirname, console, process*/
+'use strict';
+
+var express = require('express'),
+  spdy = require('spdy'),
+  bodyParser = require('body-parser'),
+  errorhandler = require('errorhandler'),
+  morgan = require('morgan'),
+  fs = require('fs'),
+  https = require('https'),
+  icsREST = require('./rest');
+
+var app = express();
+
+// app.configure ya no existe
+app.use(errorhandler());
+app.use(morgan('dev'));
+app.use(express.static(__dirname + '/public'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.disable('x-powered-by');
+
+app.use(function(req, res, next) {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Methods', 'POST, GET, PUT, PATCH, OPTIONS, DELETE');
+  res.header('Access-Control-Allow-Headers', 'origin, content-type');
+  res.header('Strict-Transport-Security', 'max-age=1024000; includeSubDomain');
+  res.header('X-Content-Type-Options', 'nosniff');
+  if (req.method == 'OPTIONS') {
+    res.send(200);
+  } else {
+    next();
+  }
+});
+
+icsREST.API.init('_service_ID_', '_service_KEY_', 'https://localhost:3000/', false);
+
+var sampleRoom;
+var pageOption = { page: 1, per_page: 100 };
+(function initSampleRoom () {
+  icsREST.API.getRooms(pageOption, function(rooms) {
+    console.log(rooms.length + ' rooms in this service.');
+    for (var i = 0; i < rooms.length; i++) {
+      if (sampleRoom === undefined && rooms[i].name === 'sampleRoom') {
+        sampleRoom = rooms[i]._id;
+        console.log('sampleRoom Id:', sampleRoom);
+      }
+      if (sampleRoom !== undefined) {
+        break;
+      }
+    }
+    var tryCreate = function(room, callback) {
+      var options = {
```

Comment:
The "transcoding: false" is not correct. You can refer to https://github.com/open-webrtc-toolkit/owt-server/blob/master/source/data_access/model/roomModel.js#L137.
Though this config may not affect room's setup.

Reply:
It looks like `transcoding: false` worked. I guess the reason is https://github.com/open-webrtc-toolkit/owt-server/blob/18a5eec64fc34b7be5c46476fed0a2987e457447/source/agent/conference/roomController.js#L47-L48 Anyway, I'll update it to comply with designed format.

Reply:
Done.